### PR TITLE
client/asset/eth: Use dex.Logger in internal node.

### PR DESCRIPTION
**Pull request by JoeGruffins**
Use dex's logger in order to more easily manage log messages from the
    internal geth node.

### CODE REVIEW

File: client/asset/eth/node.go [hunk 1]
```diff
@@ -32,8 +33,68 @@ var simnetGenesis string
...
+	return nil
+}
+
+// SetHandler updates the logger to write records to the specified handler.
+// Used during setup in geth when a logger is not supplied. Does nothing here.
+func (el *ethLogger) SetHandler(h log.Handler) {}
+
+// Trace logs at Trace level.
+func (el *ethLogger) Trace(msg string, ctx ...interface{}) {
+	el.dl.Trace(append([]interface{}{msg}, ctx...))
```
Comment (chappjc) [inline client/asset/eth/node.go:67]: Why not `el.dl.Tracef(msg, ctx...)`?

File: client/asset/eth/node.go [hunk 2]
```diff
@@ -32,8 +33,68 @@ var simnetGenesis string
...
+		s = fmt.Sprintf("%s:%s", s, v)
+	}
+	l := el.dl.SubLogger(s)
+	return &ethLogger{dl: l}
+}
+
+// GetHandler gets the handler associated with the logger. Unused in geth. Does
+// nothing and returns a nil interface here.
+func (el *ethLogger) GetHandler() log.Handler {
+	return nil
```
Comment (chappjc) [inline client/asset/eth/node.go:58]: geth is always cool with `nil`?  Don't need a no-op function like `func(*Record) error { return nil }`?

File: client/asset/eth/node.go (hunk 1 above)
Comment (chappjc) [reply client/asset/eth/node.go:67]: Oh, I just looked at the go-ethereum code and I see that `ctx` is supposed to be key/value pairs, with the entirety of the formatted message in `mgs`.  e.g. `log.Debug("Failed to decode keystore key", "path", path, "err", err)`  How should that display?

File: client/asset/eth/node.go (hunk 2 above)
Comment (JoeGruffins) [reply client/asset/eth/node.go:58]: That would probably bet better. It is totally unused in the code at present, but that could change.

File: client/asset/eth/node.go (hunk 1 above)
Comment (JoeGruffins) [reply client/asset/eth/node.go:67]: How would you like it to display? lol. I can't really see any different using this and the default logger. will get a comparison here in a sec...

Comment (chappjc) [reply client/asset/eth/node.go:67]: Yeah, it's totally up to us.  I'm just wondering how geth (the app) would print that out, so I can understand what's usually in these key-value pairs in `ctx`.  I grabbed that line randomly from the go-ethereum code base.  Seems like the intention is to show something like:

```
[DBG] GETH: Failed to decode keystore key. Details: path = /mnt/blah/key.db, err = "oh darn"
```

Comment (JoeGruffins) [reply client/asset/eth/node.go:67]: them
```
DEBUG[07-27|15:20:52.909] Removing p2p peer                        peercount=2 id=bad85b887d6606b2 duration="355.116µs" req=false err="Key missing from list - recentTxLookup"
```
us
```
2021-07-26 17:53:35.039 [TRC] ETHTEST[NODE]2021-07-26 17:53:35.039 [DBG] ETHTEST[NODE]: [Removing p2p peer peercount 2 id bad85b887d6606b246bb493caa638499f2f0576a7c2ed84fc72706b3de7ddceb duration 446.036µs req false err Key missing from list - recentTxLookup]
```
I'll make it more like theirs.

Comment (JoeGruffins) [reply client/asset/eth/node.go]: at latest commit:
```
2021-07-27 16:33:56.327 [DBG] ETHTEST[NODE]: Removing p2p peer             peercount = 3   id = a150cfb3940507cc8b340fb6b647855bf674ee82afec58f0c8426a5f354d2107   duration = 568.345µs   req = false   err = Key missing from list - recentTxLookup
```

File: client/asset/eth/node.go (hunk 2 above)
Comment (JoeGruffins) [reply client/asset/eth/node.go]: Added the noop type. Was I able to do this with an anonymous function?

Comment (chappjc) [reply client/asset/eth/node.go]: dummyHandler looks nice

File: client/asset/eth/node.go (hunk 1 above)
Comment (chappjc) [reply client/asset/eth/node.go]: Works for me.  A little space-heavy, but OK.

**Review by chappjc — APPROVED** [review]

**Pull request merged**